File: Java/BubbleSort.java
public class BubbleSort{
    public static void main(String args[]){

        int a[] = {4,5,3,8,1,2};
        int temp;

        for(int i = 0; i < a.length - 1 ; i++){

            boolean sorted = true;
            
            for(int j = 0; j < a.length - 1; j++){
                if(a[j] > a[j + 1]){
                    temp = a[j + 1];
                    a[j + 1] = a[j];
                    a[j] = temp;

                    sorted = false;

                }
            }
        
            if(sorted)
                break;
        }
    
    
        for(int i : a){
            System.out.print(i + " ");
        }
    
    
    }

}

File: Data Structures/BubbleSort.c
#include<stdio.h>
#include<stdlib.h>

void BubbleSort(int a[], int n){
   for(int k = 0; k < n - 1; k++){
        for(int i = 0; i < n - k - 1; i++){
            if(a[i] > a[i + 1]){
                int temp = a[i];
                a[i] = a[i + 1];
                a[i + 1] = temp;
            }
        }
   }
}


int main(){
    int a[] = {2, 7, 4, 1, 5, 3};

    BubbleSort(a, 6);

    for(int i = 0; i < 6; i++){
        printf(" %d", a[i]);
    }   
    printf("\n");

    return 0;
}
